File: app/src/main/java/com/joeshuff/dddungeongenerator/PDFActivity.java
package com.joeshuff.dddungeongenerator;

import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;
import androidx.appcompat.app.AppCompatDelegate;
import com.crashlytics.android.Crashlytics;
import com.github.barteksc.pdfviewer.PDFView;
import com.joeshuff.dddungeongenerator.util.AppPreferences;

import java.io.IOException;

public class PDFActivity extends AppCompatActivity {

    static String filenameKey = "PDF_filename";
    static String titleNameKey = "title_filename";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pdf_view);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        if (!getIntent().hasExtra(filenameKey) || !getIntent().hasExtra(titleNameKey)) {
            Toast.makeText(this, "Not enough info to show PDF", Toast.LENGTH_LONG).show();
            finish();
        } else {
            getSupportActionBar().setTitle(getIntent().getStringExtra(titleNameKey));

            try {
                PDFView pdfView = findViewById(R.id.pdfView);
                pdfView
                        .fromStream(getApplicationContext().getAssets().open(getIntent().getStringExtra(filenameKey)))
                        .spacing(4)
                        .nightMode(AppPreferences.INSTANCE.getDarkThemeMode() == AppCompatDelegate.MODE_NIGHT_YES)
                        .load();
            } catch (Exception e) {
                Logs.e("PDFActivity", "Error setting PDF file", e);
                Toast.makeText(this, "Something went wrong loading that PDF", Toast.LENGTH_LONG).show();
                finish();
            }
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            supportFinishAfterTransition();
            return true;
        }

        return false;
    }
}

File: app/src/main/java/com/joeshuff/dddungeongenerator/pdfviewing/PDFActivity.kt
package com.joeshuff.dddungeongenerator.pdfviewing

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.joeshuff.dddungeongenerator.util.Logs
import com.joeshuff.dddungeongenerator.R
import com.joeshuff.dddungeongenerator.util.AppPreferences.pdfIsDark
import kotlinx.android.synthetic.main.activity_pdf_view.*

open class PDFActivity : AppCompatActivity() {

    companion object {
        var filenameKey = "PDF_filename"
        var titleNameKey = "title_filename"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pdf_view)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        if (!intent.hasExtra(filenameKey) || !intent.hasExtra(titleNameKey)) {
            Toast.makeText(this, "Not enough info to show PDF", Toast.LENGTH_LONG).show()
            finish()
        } else {
            supportActionBar?.title = intent.getStringExtra(titleNameKey)

            try {
                pdfView
                        .fromStream(applicationContext.assets.open(intent.getStringExtra(filenameKey)))
                        .spacing(4)
                        .nightMode(pdfIsDark)
                        .load()
            } catch (e: Exception) {
                Logs.e("PDFActivity", "Error setting PDF file", e)
                Toast.makeText(this, "Something went wrong loading that PDF", Toast.LENGTH_LONG).show()
                finish()
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.pdf_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                supportFinishAfterTransition()
                return true
            }
            R.id.pdf_day_night_toggle -> {
                pdfIsDark = !pdfIsDark
                pdfView.setNightMode(pdfIsDark)
                pdfView.requestLayout()
                return true
            }
        }

        return false
    }
}